test/e2e/conformance/validatingadmissionpolicy_test.go

```diff
@@ -291,6 +292,17 @@ func TestValidatingAdmissionPolicyCrossWorkspaceAPIBinding(t *testing.T) {
 	_, err = kcpClusterClient.Cluster(sourcePath).ApisV1alpha2().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
 	require.NoError(t, err)
 
+	t.Logf("Creating default namespace in target workspace")
+	defaultNS := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+	}
+	_, err = kubeClusterClient.Cluster(targetPath).CoreV1().Namespaces().Create(ctx, defaultNS, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		require.NoError(t, err, "failed to create default namespace")
+	}
```

Comment:
This and the change below is more a bandaid instead of a fix. It'd be better to address the root cause of the missing namespace.

Looking at this code this has been a problem in the past: https://github.com/kcp-dev/kcp/blob/c885efca61e51495bb8c53d8c07b08455c22a598/test/e2e/reconciler/workspacedeletion/controller_test.go#L103-L122

Instead of waiting on the namespace it would be preferable to inspect why the workspace phase is set to ready before the default namespace is present.